pkg/helmexec/exec.go

```diff
@@ -418,21 +418,15 @@ func (helm *execer) Fetch(chart string, flags ...string) error {
 	return err
 }
 
-func (helm *execer) ChartPull(chart string, flags ...string) error {
+func (helm *execer) ChartPull(chart string, path string, flags ...string) error {
+	var helmArgs []string
 	helm.logger.Infof("Pulling %v", chart)
-	helm.logger.Infof("Exporting %v", chart)
 	helmVersionConstraint, _ := semver.NewConstraint(">= 3.7.0")
-	var helmArgs []string
 	if helmVersionConstraint.Check(&helm.version) {
+		// in the 3.7.0 version, the chart pull has been replaced with helm pull
+		// https://github.com/helm/helm/releases/tag/v3.7.0
 		ociChartURL, ociChartTag := resolveOciChart(chart)
-		tempDir, err := os.MkdirTemp("", "chart*")
-		if err != nil {
-			return err
-		}
-		defer func() {
-			_ = os.RemoveAll(tempDir)
-		}()
-		helmArgs = []string{"fetch", ociChartURL, "--version", ociChartTag, "--destination", tempDir}
+		helmArgs = []string{"pull", ociChartURL, "--version", ociChartTag, "--destination", path}
 	} else {
 		helmArgs = []string{"chart", "pull", chart}
```

Comment:
@yxxhero FYI, this is the legacy code path for helm 3.7.x or less. We no longer support 3.7.x since https://github.com/helmfile/helmfile/pull/286 and this should probably be safe enough to be removed now.

Reply:
yes. you are right. @mumoshu